### ISSUE TITLE
Test indexing against a wide range of gems

### ISSUE THREAD

**Issue by andyw8** [Shopify/ruby-lsp#2299]
Similar to what Prism does here: https://github.com/ruby/prism/blob/main/rakelib/lex.rake

**Comment by andyw8** [comment]
Top 100 done: https://github.com/Shopify/ruby-lsp/pull/2330

Next step will be cover all of RubyGems.org.

**Comment by vinistock** [comment]
Question: is it really necessary to go beyond the 100 top gems?

We run indexing on CI for all of our fixtures, all of Prism fixtures, the Ruby LSP's own codebase and the 100 top gems.

I wonder if this is already enough to catch most problems and if expanding to all gems would actually bring any benefit (it will take a long time to index every single gem).

**Comment by st0012** [comment]
Yeah I think we'll only get diminishing return to chase full rubygems.org coverage, with a greater investment on building the infra for it. I feel it's ok to close this first, and then if we do see more indexing issues from smaller gems, we reopen the issue for further discussion.
